phantazm-commons/src/main/java/com/github/phantazmnetwork/commons/LogicUtils.java

```diff
@@ -0,0 +1,35 @@
+package com.github.phantazmnetwork.commons;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+/**
+ * Utilities for controlling program flow and logic.
+ */
+public final class LogicUtils {
```

Comment:
this in general is a meh name, but I don't have a much better suggestion

Reply:
could be ControlFlowUtils but that's worse, i'll leave it for now